android/src/main/java/com/expensify/livemarkdown/RangeSplitter.java

```diff
@@ -0,0 +1,44 @@
+package com.expensify.livemarkdown;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RangeSplitter {
+  public static void splitRangesOnEmojis(List<MarkdownRange> markdownRanges, String type) {
+    List<MarkdownRange> emojiRanges = new ArrayList<>();
+    List<MarkdownRange> oldRanges = new ArrayList<>(markdownRanges);
```

Comment:
Let's explicitly mark this type as `ArrayList` to guarantee that reads are O(1).
```suggestion
    ArrayList<MarkdownRange> oldRanges = new ArrayList<>(markdownRanges);
```